UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService.UnitTests/Validation/ConfigValidatorTests.cs

```diff
@@ -0,0 +1,336 @@
+        {
+            BessConfig bessConfig = new()
+            {
+                Name = "BES-1",
+                ExchangeSetStandard = exchangeSetStandard,
+                EncCellNames = new List<string> { "GB123456", "GB234567", "GB*" },
+                Frequency = "15 16 2 2 *",
+                Type = type,
+                KeyFileType = keyFileType,
+                AllowedUsers = new List<string> { "User1", "User2" },
+                //No Allowed Users Group
+                Tags = new List<Tag> { new() { Key = "key1", Value = "value1" }, new() { Key = "key2", Value = "value2" } },
+                ReadMeSearchFilter = readMeSearchFilter,
+                BatchExpiryInDays = 1,
+                IsEnabled = isEnabled
+            };
+
+            TestValidationResult<BessConfig> result = configValidator.TestValidate(bessConfig);
+            result.Errors.Count.Should().Be(0);
+        }
+
+        [Test]
+        [TestCase("1")]
+        public void WhenConfigContainsInvalidIsEnabledAttribute_ThenThrowValidationErrorForIsEnabledOnly(string isEnabled)
+        {
+            var bessConfig = new BessConfig
+            {
+                Name = null,
+                ExchangeSetStandard = null,
+                EncCellNames = new List<string>(),
+                Frequency = "",
+                Type = "",
+                KeyFileType = "",
+                AllowedUsers = new List<string>(),
+                AllowedUserGroups = new List<string>(),
+                Tags = null,
+                ReadMeSearchFilter = "",
+                BatchExpiryInDays = 0,
+                IsEnabled = isEnabled
+            };
+
+            TestValidationResult<BessConfig> result = configValidator.TestValidate(bessConfig);
+
+            result.ShouldHaveValidationErrorFor(x => x.IsEnabled)
+                .WithErrorMessage("Attribute is missing or value not provided. Expected value is either Yes or No.");
+        }
+
+        [Test]
+        public void WhenIsEnabledYesAndConfigContainsInvalidAttributes_ThenThrowValidationError()
+        {
+            var bessConfig = new BessConfig
+            {
+                Name = null,
+                ExchangeSetStandard = null,
+                EncCellNames = new List<string>(),
+                Frequency = null,
+                Type = null,
+                KeyFileType = null,
+                AllowedUsers = new List<string>(),
+                AllowedUserGroups = new List<string>(),
+                Tags = null,
+                ReadMeSearchFilter = "",
+                BatchExpiryInDays = 0,
+                IsEnabled = "yes"
+            };
+
+            TestValidationResult<BessConfig> result = configValidator.TestValidate(bessConfig);
+
+            result.ShouldHaveValidationErrorFor(x => x.Name)
+                .WithErrorMessage("Attribute is missing or value not provided");
```

Comment:
As this validation message text is same for different validation can we make constant string and use that across ?
```suggestion
                .WithErrorMessage("Attribute is missing or value is not provided");
```

Reply:
Resolved. Thanks